code/modules/power/lighting.dm

```diff
@@ -129,58 +134,89 @@
 		new /obj/item/stack/sheet/metal(loc, sheets_refunded)
 	qdel(src)
 
+/**
+  * # Small light fixture frame
+  *
+  * Incomplete light bulb fixture
+  *
+  * Become a [Small light fixture] when completed
+  */
 /obj/machinery/light_construct/small
 	name = "small light fixture frame"
 	desc = "A small light fixture under construction."
 	icon = 'icons/obj/lighting.dmi'
 	icon_state = "bulb-construct-stage1"
-	anchored = 1
+	anchored = TRUE
 	layer = 5
 	stage = 1
 	fixture_type = "bulb"
 	sheets_refunded = 1
 
 
-// the standard tube light fixture
+/**
+  * # Light fixture
+  *
+  * The standard light tube fixture
+  */
 /obj/machinery/light
 	name = "light fixture"
 	icon = 'icons/obj/lighting.dmi'
-	var/base_state = "tube"		// base description and icon_state
+	var/base_state = "tube" // Base description and icon_state
 	icon_state = "tube1"
 	desc = "A lighting fixture."
 	anchored = 1
-	layer = 5  					// They were appearing under mobs which is a little weird - Ostaf
+	layer = 5
 	max_integrity = 100
 	use_power = ACTIVE_POWER_USE
 	idle_power_usage = 2
 	active_power_usage = 20
 	power_channel = LIGHT //Lights are calc'd via area so they dont need to be in the machine list
-	var/on = FALSE					// 1 if on, 0 if off
+	/// Is the light on or off?
+	var/on = FALSE
 	var/on_gs = 0
```

Comment:
Could you document this var? If you know what it does that is.

Reply:
No idea what that does, it's only used here:
```
if(on != on_gs)
    on_gs = on
```
I couldn't figure out why it needs to be there, so I just left it until I had time to test.

Reply:
Honestly I can't even follow what its suppose to do either. I would remove it and then test lights, see what happens. Or just leave it alone.  :smile:

Reply:
It is here to check if the state changed between `update` calls. If update is called when the light is on and before was on as well then it won't do the power stuff below.